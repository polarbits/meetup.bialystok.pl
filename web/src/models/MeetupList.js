import { types } from 'mobx-state-tree'
import { number } from 'mobx-state-tree/dist/internal';

const data = {
  'title': 'Meetjs #16',
  'description': 'lorem ipsum',
  'image': '/img/meetup.jpg'
}

export const Meetup = types.model({
  title: types.string,
  description: types.string,
  image: types.optional(types.string, '/img/default.jpg'),
  price: types.number
}).actions(self => ({
  changeTitle(newTitle) {
    self.title = newTitle
  },
}))

export const MeetupList = types.model({
  items: types.optional(types.array(Meetup), [])
}).actions(self => ({
  add(item) {
    self.items.push(item)
  }
}))
.views(self => ({
  get totalPrice() {
    return self.items.reduce((sum, entry) => sum + entry.price, 0)
  }
}))