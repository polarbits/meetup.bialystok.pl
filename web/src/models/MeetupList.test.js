import { getSnapshot, onSnapshot, onPatch } from 'mobx-state-tree'
import { reaction } from "mobx"
import { MeetupList, Meetup } from './MeetupList' 
it('can create a meetup', () => {
  const item = Meetup.create({
    'title': 'Meetjs #16',
    'description': 'lorem ipsum',
    'image': '/img/meetup.jpg'
  })

  expect(item.title).toBe('Meetjs #16')
  item.changeTitle('Meetjs #17')
  expect(item.title).toBe('Meetjs #17')
})

it('can create a meetup list', () => {
  const list = MeetupList.create({
    items: [{
      'title': 'Meetjs #16',
      'description': 'lorem ipsum',
      'image': '/img/meetup.jpg',
      'price': 10
    },
    {
      'title': 'Meetjs #16',
      'description': 'lorem ipsum',
      'image': '/img/meetup.jpg',
      'price': 20
    },{
      'title': 'Meetjs #16',
      'description': 'lorem ipsum',
      'image': '/img/meetup.jpg',
      'price': 30
    }
    ]
  })
  expect(list.items.length).toBe(3)
})

it ('can add new items', () => {
  const list = MeetupList.create()
  const patches = []
  onPatch(list, patch => {
    patches.push(patch)
  })
  list.add({
    title: 'meetjs',
    description: 'lorem ipsum dolor',
    'price': 30
  })
  
  list.items[0].changeTitle('meetjs')

  expect(patches).toMatchSnapshot()
})

it('can calculate', () => {
  const list = MeetupList.create({ items: [
    {
      'title': 'Meetjs #16',
      'description': 'lorem ipsum',
      'image': '/img/meetup.jpg',
      'price': 10
    },
    {
      'title': 'Meetjs #16',
      'description': 'lorem ipsum',
      'image': '/img/meetup.jpg',
      'price': 20
    }
  ]})
  expect(list.totalPrice).toBe(30)
})
