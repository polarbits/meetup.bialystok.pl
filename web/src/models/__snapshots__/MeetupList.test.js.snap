// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new items 1`] = `
Object {
  "items": Array [
    Object {
      "description": "lorem ipsum dolor",
      "image": "/img/default.jpg",
      "title": "meetjs",
    },
  ],
}
`;

exports[`can add new items 2`] = `
Array [
  Object {
    "items": Array [
      Object {
        "description": "lorem ipsum dolor",
        "image": "/img/default.jpg",
        "title": "meetjs",
      },
    ],
  },
]
`;
